pytorch/imagerecognition/openmpi-cifar10-resnet20-all-reduce-scaling/main.py

```diff
@@ -169,19 +169,30 @@ def train_loop(run_id, dataset_dir, ckpt_run_dir, output_dir,
             json.dump(val_stats, f)
 
 
-def main(run_id, dataset_dir, ckpt_run_dir, output_dir, validation_only=False,
+def main(run_id, dataset_dir, ckpt_run_dir, output_dir, rank, world_size, backend, hosts, validation_only=False,
```

Comment:
Not sure here if `world_size`, `rank` and `hosts` are necessary. Usually they are deduced using the `torch.distributed` package:
```
rank = dist.get_rank()
world_size = dist.get_world_size()
```

Same goes for all `main` functions

Reply:
woldsize can also be calculated using `len(hosts)`

Reply:
They are decoded only if they are environment variables. I think when we use Mpi, Openmpi somehow sets them and knows how to read them, but I tried with Gloo and it does not work like that. So we either have to pass them like this or set them as environment variables. I don't know how to do this, so please let me know if you have any suggestions.